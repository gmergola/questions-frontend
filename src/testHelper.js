import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

/**
 * Helper function to modify the render function in React Testing Library
 * by wrapping rendered components inside of a router.
 *
 * Taken from https://testing-library.com/docs/example-react-router
 *
 * @param {Component} component - React component we want to render
 * @param {Object} param1 - object with route and history keys
 */

export function renderWithRouter(component,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  const Wrapper = ({ children }) => (
    <Router history={history}>{children}</Router>
  );
  return {
    ...render(component, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  };
}