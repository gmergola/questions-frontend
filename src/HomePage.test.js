import React from "react";
import { renderWithRouter } from './testHelper';
import Home from "./Homepage";


it("matches snapshot", function() {
  const { asFragment } = renderWithRouter(<Home />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function() {
  const { asFragment } = renderWithRouter(<Home />);
  expect(asFragment()).toMatchSnapshot();
});
