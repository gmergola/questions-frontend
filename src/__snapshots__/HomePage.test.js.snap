// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="HomePage-container"
  >
    <button
      class="HomePage-directions-button"
    >
      <b>
        How to play?
      </b>
    </button>
    <div
      class="HomePage-title-container"
    >
      <div
        class="HomePage-play"
      >
        Let's play
      </div>
      <div
        class="HomePage-title"
      >
        Questions!
      </div>
    </div>
    <p>
      loading...
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="HomePage-container"
  >
    <button
      class="HomePage-directions-button"
    >
      <b>
        How to play?
      </b>
    </button>
    <div
      class="HomePage-title-container"
    >
      <div
        class="HomePage-play"
      >
        Let's play
      </div>
      <div
        class="HomePage-title"
      >
        Questions!
      </div>
    </div>
    <p>
      loading...
    </p>
  </div>
</DocumentFragment>
`;
